### PR TITLE
better description for some modules

### LINKED ISSUE #944: add module description in refinement, topology, and contactmap modules
the automatically generated haddock3 documentation renders the description present in the __init__ file of each module. Therefore the modules that do not have it should be updated with a brief description.

### PR DESCRIPTION
**Pull request by VGPReys**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and that you comply with the following criteria:

- [x] You have sticked to Python. Please talk to us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [x] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [ ] You structured the code into small functions as much as possible. You can use classes if there is a (state) purpose
- [x] Your code follows our coding style
- [ ] You wrote tests for the new code
- [x] `tox` tests pass. *Run `tox` command inside the repository folder*
- [x] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [x] PR does not add any dependencies, unless permission granted by the HADDOCK team
- [x] PR does not break licensing
- [X] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

Closes #944

### CODE REVIEW

File: src/haddock/modules/topology/topoaa/__init__.py [hunk 1]
```diff
@@ -1,4 +1,27 @@
...
+- Build the topology
+
+This module is a pre-requisite to run any downstream modules using CNS.
+Indeed, having access to parameters and topology is mandatory for any kind
+of MD related tasks.
+Therefore this is the reason why the module [topoaa] is often used as first
+module in a workflow.
+
+Note that for non-standard bio-molecules
+(appart from standard amino-acids, some modified ones, DNA, RNA, ions
```
Comment (mgiulini) [inline src/haddock/modules/topology/topoaa/__init__.py:19]:
```suggestion
(apart from standard amino-acids, some modified ones, DNA, RNA, ions
```

**Review by mgiulini — APPROVED** [review]

File: src/haddock/modules/analysis/contactmap/__init__.py [hunk 2]
```diff
@@ -1,4 +1,18 @@
...
+the contacts observed in the input complexes.
+
+If complexes are clustered, the analysis of contacts will be performed
+based on all structures from each cluster.
+
+**Heatmaps** are describing the probability of contacts (<5A) between two
+residues (both intramolecular and intermolecular).
+
+**Chordcharts** are describing only intermolecular contacts in circles,
+connecting with *chords* the two resiudes that are contacting.
```
Comment (mgiulini) [inline src/haddock/modules/analysis/contactmap/__init__.py:13]:
```suggestion
connecting with *chords* the two residues that are contacting.
```

File: src/haddock/modules/refinement/emref/__init__.py [hunk 3]
```diff
@@ -1,4 +1,13 @@
-"""Energy minimization refinement with CNS."""
+"""Energy minimization refinement with CNS.
+
+The `[emref]` module refine the input complexes by energy minimization using
+the conjugate gradient method develped by
```
Comment (mgiulini) [inline src/haddock/modules/refinement/emref/__init__.py:4]:
```suggestion
the conjugate gradient method developed by
```

**Review by mgiulini — DISMISSED** [review]

**Review by rvhonorato — APPROVED** [review]

File: src/haddock/modules/refinement/flexref/__init__.py [hunk 4]
```diff
@@ -1,4 +1,16 @@
-"""Flexible refinement with CNS."""
+"""Flexible refinement with CNS.
+
+The `[flexref]` module (also known as `it1` stage in HADDOCK2.X series),
+is a semi-flexible simulated annealing (SA) protocol in torsion angle space.
```
Comment (amjjbonvin) [inline src/haddock/modules/refinement/flexref/__init__.py:4]: protocol based on molecular dynamics in torsion angle space

File: src/haddock/modules/refinement/flexref/__init__.py [hunk 5]
```diff
@@ -1,4 +1,16 @@
...
+
+This semi-flexible SA consists of several stages:
+- High temperature rigid body search
+- Rigid body SA
+- Semi-flexible SA with flexible side-chains at the interface
+- Semi-flexible SA with fully flexible interface (both backbone and side-chains)
+
```
Comment (amjjbonvin) [inline src/haddock/modules/refinement/flexref/__init__.py:11]: By default, only the interface regions is treated as flexible. It is automatically defined based on the intermolecular contacts made. It is also possible to manually define the semi-flexible regions, and also define fully flexible regions that are allowed to move throughout the entire protocol from the high temperature rigid body molecular dynamics on.

File: src/haddock/modules/refinement/flexref/__init__.py [hunk 6]
```diff
@@ -1,4 +1,16 @@
-"""Flexible refinement with CNS."""
+"""Flexible refinement with CNS.
+
+The `[flexref]` module (also known as `it1` stage in HADDOCK2.X series),
+is a semi-flexible simulated annealing (SA) protocol in torsion angle space.
+
+This semi-flexible SA consists of several stages:
+- High temperature rigid body search
```
Comment (amjjbonvin) [inline src/haddock/modules/refinement/flexref/__init__.py:7]: Replace "search" by "molecular dynamics"

File: src/haddock/modules/refinement/mdref/__init__.py [hunk 7]
```diff
@@ -1,4 +1,24 @@
...
+
+The `[mdref]` module (also known as `itw` in HADDOCK2.X series), is a small
+molecular dynamics simulation, in cartesian space, using explicit solvent.
+
+A layer of solvent (8A for water, 12.5A for DMSO) is generated around
+surface residues.
+
+The `mdref` protocol is composed of 4 sequential steps:
+- Short energy minimization
+- 3 stages of water heating to reach 300K (at 100, 200 and 300K)
```
Comment (amjjbonvin) [inline src/haddock/modules/refinement/mdref/__init__.py:11]: water heating by molecular dynamics

File: src/haddock/modules/refinement/mdref/__init__.py [hunk 8]
```diff
@@ -1,4 +1,24 @@
...
+- A series of steps at 300K.
```
Comment (amjjbonvin) [inline src/haddock/modules/refinement/mdref/__init__.py:12]: Molecular dynamics at 300K

File: src/haddock/modules/refinement/mdref/__init__.py [hunk 9]
```diff
@@ -1,4 +1,24 @@
...
+- 3 stages of water cooling, to reach 100K (at 300, 200 and 100K)
```
Comment (amjjbonvin) [inline src/haddock/modules/refinement/mdref/__init__.py:13]: water cooling by molecular dynamics

File: src/haddock/modules/topology/topoaa/__init__.py [hunk 10]
```diff
@@ -1,4 +1,27 @@
-"""Create and manage CNS all-atom topology."""
+"""Create and manage CNS all-atom topology.
+
+The `[topoaa]` module is dedicated to the generation of CNS compatible
+parameters (.param) and topologies (.psf) for each of the input structures.
+
+It will:
+- Detect missing atoms, including hydrogens
+- Re-build them when missing
+- Obtain MD parameters
```
Comment (amjjbonvin) [inline src/haddock/modules/topology/topoaa/__init__.py:9]: Remove: Obtain MD parameters

File: src/haddock/modules/topology/topoaa/__init__.py [hunk 11]
```diff
@@ -1,4 +1,27 @@
...
+- Build the topology
```
Comment (amjjbonvin) [inline src/haddock/modules/topology/topoaa/__init__.py:10]: Build and write out topologies (psf) and coordinates (pdb) files

File: src/haddock/modules/topology/topoaa/__init__.py [hunk 12]
```diff
@@ -1,4 +1,27 @@
...
+
+This module is a pre-requisite to run any downstream modules using CNS.
+Indeed, having access to parameters and topology is mandatory for any kind
+of MD related tasks.
```
Comment (amjjbonvin) [inline src/haddock/modules/topology/topoaa/__init__.py:14]: EM/MD

File: src/haddock/modules/topology/topoaa/__init__.py [hunk 13]
```diff
@@ -1,4 +1,27 @@
...
+parameters (.param) and topologies (.psf) for each of the input structures.
+
+It will:
+- Detect missing atoms, including hydrogens
+- Re-build them when missing
+- Obtain MD parameters
+- Build the topology
+
+This module is a pre-requisite to run any downstream modules using CNS.
+Indeed, having access to parameters and topology is mandatory for any kind
```
Comment (amjjbonvin) [inline src/haddock/modules/topology/topoaa/__init__.py:13]: Remove: Indeed,

File: src/haddock/modules/topology/topoaa/__init__.py [hunk 14]
```diff
@@ -1,4 +1,27 @@
...
+of MD related tasks.
+Therefore this is the reason why the module [topoaa] is often used as first
+module in a workflow.
+
+Note that for non-standard bio-molecules
+(apart from standard amino-acids, some modified ones, DNA, RNA, ions
+and carbohydrates ... see [detailed list of supported molecules]
+(https://rascar.science.uu.nl/haddock2.4/library)),
+such as small-molecules, parameters and topology must be obtained and provided,
```
Comment (amjjbonvin) [inline src/haddock/modules/topology/topoaa/__init__.py:22]: provided by the user

File: src/haddock/modules/topology/topoaa/__init__.py [hunk 15]
```diff
@@ -1,4 +1,27 @@
...
+This module is a pre-requisite to run any downstream modules using CNS.
+Having access to parameters and topology is mandatory for any kind
+of EM/MD related tasks.
+Therefore this is the reason why the module [topoaa] is often used as first
+module in a workflow.
+
+Note that for non-standard bio-molecules
+(apart from standard amino-acids, some modified ones, DNA, RNA, ions
+and carbohydrates ... see [detailed list of supported molecules]
+(https://rascar.science.uu.nl/haddock2.4/library)),
```
Comment (rvhonorato) [inline src/haddock/modules/topology/topoaa/__init__.py:20]:
```suggestion
(https://wenmr.science.uu.nl/haddock2.4/library)),
```

**Review by amjjbonvin — APPROVED** [review]
Do implement Rodrigo's suggestion for the URL

File: src/haddock/modules/refinement/emref/__init__.py [hunk 16]
```diff
@@ -1,4 +1,13 @@
-"""Energy minimization refinement with CNS."""
+"""Energy minimization refinement with CNS.
+
+The `[emref]` module refine the input complexes by energy minimization using
+the conjugate gradient method developed by
+[Powell](https://en.wikipedia.org/wiki/Powell%27s_method),
```
Comment (rvhonorato) [inline src/haddock/modules/refinement/emref/__init__.py:5]: The syntax for links is 
```rst
`Powell <https://en.wikipedia.org/wiki/Powell%27s_method>`_
```

File: src/haddock/modules/refinement/emref/__init__.py [hunk 17]
```diff
@@ -1,4 +1,13 @@
-"""Energy minimization refinement with CNS."""
+"""Energy minimization refinement with CNS.
+
+The `[emref]` module refine the input complexes by energy minimization using
```
Comment (rvhonorato) [inline src/haddock/modules/refinement/emref/__init__.py:3]: The syntax for code block is 

```rst
``[emref]``
```

**Review by rvhonorato — CHANGES_REQUESTED** [review]
Building the docs to check the changes, it seems that this esoteric documentation building pipeline renders the docstrings as `rst` and then `html`, but your changes are in markdown.

So they are not rendered properly:

Schermafbeelding 2024-07-24 100302

Have a look here for the proper syntax: https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html

**Comment by amjjbonvin** [comment]
The current syntax is a proper markdown syntax, but I guess we are speaking here about python code/comments, correct?

**Comment by rvhonorato** [comment]
> The current syntax is a proper markdown syntax, but I guess we are speaking here about python code/comments, correct?


?

**Comment by amjjbonvin** [comment]
I mean:

> [Powell](https://en.wikipedia.org/wiki/Powell%27s_method)

is a correct markdown syntax for a link.

**Comment by rvhonorato** [comment]
It is, yes - however the changes in this PR should be in RST

File: src/haddock/modules/refinement/emref/__init__.py [hunk 18]
```diff
@@ -1,4 +1,13 @@
-"""Energy minimization refinement with CNS."""
+"""Energy minimization refinement with CNS.
+
+The ``[emref]`` module refine the input complexes by energy minimization using
+the conjugate gradient method developed by
+`Powell <https://en.wikipedia.org/wiki/Powell%27s_method>`_,
```
Comment (amjjbonvin) [inline src/haddock/modules/refinement/emref/__init__.py:5]: Seems random to add a link to Powel and not other things. I would  remove it and simply write: "by conjugate gradient energy minimisation."

**Review by amjjbonvin — APPROVED** [review]

**Review by mgiulini — APPROVED** [review]

**Pull request merged**